context/context.go

```diff
@@ -202,3 +256,60 @@ func (c *ctx) returnToPool() {
 	c.Reset()
 	c.pool.Put(c)
 }
+
+func (c *ctx) NewChildContext() Context {
+	var child Context
+	if c.pool == nil {
+		child = NewContext()
+	} else {
+		child = c.pool.Get()
+	}
+
+	child.SetParentCtx(c)
+
+	return child
+}
+
+func (c *ctx) SetParentCtx(parentCtx Context) {
+	// check to see if parent exists?
+	c.parent = parentCtx
+}
+
+func (c *ctx) ParentCtx() Context {
+	c.RLock()
+	defer c.RUnlock()
+	return c.parent
+}
+
+func (c *ctx) StartSpan(name string) (Context, opentracing.Span, bool) {
```

Comment:
Can we rename this to `StartTraceSpan` since then the usage is clear it's for a trace?